Trees/BinarySearchTree.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef BINARY_SEARCH_TREE
+#define BINARY_SEARCH_TREE
+
+struct Node {
+    int value;
+    Node* left_child;
+    Node* right_child;
+};
+
+Node* get_node();
+
+class BinarySearchTree {
+private:
+    Node* root;
+
+    void traversal_inorder_helper(Node* tree);
+    void traversal_preorder_helper(Node* tree);
+    void traversal_postorder_helper(Node* tree);
```

Comment:
I don't think there's a need for these 3 functions. They can be removed (from the _cpp_ file as well)